django/db/models/expressions.py

```diff
@@ -547,6 +550,58 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None,
         return ResolvedOuterRef(self.name)
 
 
+class SliceableF(F):
+    """
+    An object that can add slicing notation to F expression.
+
+    Object resolves the column on which the slicing is applied, and then
+    applies the slicing if possible.
+    """
+    def __init__(self, f_obj, slice_obj):
+        if not isinstance(slice_obj, (int, slice)):
+            raise TypeError
```

Comment:
Please provide a message, e.g., `TypeError('Slicing must be of type made either int or slice.')` and write a test as well similar.